ChopChop/ViewModels/Ingredients/IngredientFormViewModel.swift

```diff
@@ -0,0 +1,55 @@
+import SwiftUI
+
+class IngredientFormViewModel: ObservableObject {
+    private(set) var ingredient: Ingredient
+    let isEdit: Bool
+
+    @Published var selectedType: QuantityType
+    @Published var inputName: String
+    @Published var image: UIImage
+
+    @Published var isShowingPhotoLibrary = false
+    var pickerSourceType: UIImagePickerController.SourceType = .photoLibrary
+
+    init(edit ingredient: Ingredient) {
+        self.ingredient = ingredient
+        self.isEdit = true
+
+        self.selectedType = ingredient.quantityType
+        self.inputName = ingredient.name
+        self.image = ingredient.image ?? UIImage()
+    }
+
+    init() throws {
+        self.ingredient = try Ingredient(name: "temporary", type: .count)
+        self.isEdit = false
+
+        self.selectedType = .count
+        self.inputName = ""
+        self.image = UIImage()
+    }
+
+    var areFieldsValid: Bool {
+        !inputName.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
+    }
+
+    func save() throws {
+        let storageManager = StorageManager()
```

Comment:
Should we just instantiate 1 StorageManager throughout the app?

Reply:
I think we might need to display a Label under each form field to display the errors here instead of propagating the errors